src/mock.rs

```diff
@@ -0,0 +1,41 @@
+// This file is part of Astarte.
+//
+// Copyright 2023 SECO Mind Srl
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+//! Mocks for the Astarte Device SDK.
+
+use mockall::mock;
+use rumqttc::{ClientError, ConnectionError, Event, MqttOptions, QoS};
+
+mock!(
+    pub AsyncClient {
+        // NOTE: This is mocked for compatibility, but it will be hard to test
```

Comment:
What does this comment mean?
The call to `new ` is not in a difficult-to-test spot. Mocking `pairing::get_transport_config` would be sufficient to test the arguments and the return values of this function.
Even if this function was hard to test, this happens often, no need to specify it above each function.

Reply:
We would need to serialize the tests or have a lock on the tests, since this is a static method. See the mock all [synchronization example](https://github.com/asomers/mockall/blob/master/mockall/examples/synchronization.rs).

I put the comment since the function is never actually used and would be hard to use in the tests (hard to test).

It is called in the AstarteSDK::new, and I didn't write the test for the above reason and because it's a pretty straight forward function.

Reply:
Comment removed

Reply:
True, the effort needed to write the test since it's a very short and straightforward function.
However, I would still remove the comment.